neqo-http3/src/server_events.rs

```diff
@@ -53,24 +78,22 @@ impl ClientRequestStream {
     /// It may return `InvalidStreamId` if a stream does not exist anymore.
     pub fn set_response(&mut self, headers: &[Header], data: &[u8]) -> Res<()> {
         qinfo!([self], "Set new response.");
-        self.handler
-            .borrow_mut()
-            .set_response(self.stream_id, headers, data)
+        self.stream_handler.handler.borrow_mut().set_response(
+            self.stream_handler.stream_id,
+            headers,
+            data,
+        )
```

Comment:
This part is HTTP-specific, so it can go here, but you might want to think about how much this object needs to manipulate the contents of the stream_handler's handler in order to get things done.